src/snitch_main.cpp

```diff
@@ -2,14 +2,22 @@
 #include "snitch/snitch_registry.hpp"
 
 #if SNITCH_DEFINE_MAIN
-SNITCH_EXPORT int main(int argc, char* argv[]) {
-    std::optional<snitch::cli::input> args = snitch::cli::parse_arguments(argc, argv);
-    if (!args) {
-        return 1;
+namespace snitch {
+int main(int argc, char* argv[]) {
+    if constexpr (snitch::is_enabled) {
+        std::optional<snitch::cli::input> args = snitch::cli::parse_arguments(argc, argv);
+        if (!args) {
+            return 1;
+        }
+        snitch::tests.configure(*args);
+        return snitch::tests.run_tests(*args) ? 0 : 1;
+    } else {
+        return 0;
     }
+}
+} // namespace snitch
 
-    snitch::tests.configure(*args);
-
-    return snitch::tests.run_tests(*args) ? 0 : 1;
+SNITCH_EXPORT int main(int argc, char* argv[]) {
+    return snitch::main(argc, argv);
```

Comment:
This new `snitch::main()` isn't being advertised in any header, so people can't use it unless they use the header-only build. I've added a header for it and moved it out of the `SNITCH_DEFINE_MAIN` block. Let me know if that wasn't what you intended.